rfcs/2023-01-06-layout.md

```diff
@@ -0,0 +1,461 @@
+  flexDirection="column"
+  alignItems="center"
+  marginTop="spacing.1"
+  marginBottom="spacing.2"
+  paddingX="spacing.1"
+>
+  <Button>Button 1</Button>
+  <Button sx={{ marginTop: 'spacing.1' }}>Button 2</Button>
+  <Button sx={{ marginTop: 'spacing.2' }}>Button 3</Button>
+</Box>
+```
+
+`sx` prop will be the preferred choice for minor adjustments that can be done directly on the component without a need for creating extra `Box` wrapper _(which also keeps the DOM less nested and easier to read)_. We won't have layout props support for presentational components eg. `Button`, `Alert`, etc. This also gives a nicer separation between other props and layout specific props. Later on, we might want to expand the scope of `sx` to cover things beyond layout if need arises.
+
+As a tradeoff, `sx` prop is pretty dangerous in terms of the flexibility it provides. It gives power to customize any component to beyond recognition, for example you could customize the padding that comes on `Button`. In light of this capability we should support only a restricted set of properties to begin with _(which can't have adverse effect on look and feel of components)_.
+
+`sx` should be the preferred choice for minor adjustments only. You may note `padding*` is skipped since it can break existing padding of blade components. Supported restricted properties:
+
+| Category            | Props                                                                          | Notes                                                                                                                                                               |
+| ------------------- | ------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Spacing             | `margin*`                                                                      | `marginTop`, `marginLeft`, `marginRight`, `marginBottom`, `marginX`, `marginY` (React Native `marginHorizontal`, `marginVertical` mapped to these)                  |
+| Flexbox             | `alignSelf`, `justifySelf`, `order`                                            | Only restricted set of properties applicable on flex children. Note: `justifySelf` doesn't work for flex layout but it can be used for block layout (web only).     |
+| Position            | `position`, `zIndex`, `top`, etc.                                              | [List of props](https://github.com/styled-system/styled-system/blob/master/docs/table.md#position) referenced from styled-system.                                   |
+| Grid **(web only)** | `gridColumn*`, `gridRow*`, `gridArea`, `justifySelf`, `alignSelf`, `placeSelf` | Only restricted set of properties applicable on [grid children](https://css-tricks.com/snippets/css/complete-guide-grid/#aa-properties-for-the-childrengrid-items). |
```

Comment:
do these props make sense on presentational components? usually you'll use this prop mostly for spacing and for everything else we have other primitive components right?

Reply:
Yup, we would need the children specific props such as `gridColumnStart`, `gridRowEnd`, etc. [Ref](https://css-tricks.com/snippets/css/complete-guide-grid/#aa-properties-for-the-childrengrid-items)